guides/advancing-proofs/simplifications-guide.md

```diff
@@ -0,0 +1,444 @@
+# A Comprehensive Guide to Writing Simplifications for Kontrol
+
+## What are simplifications?
+
+Symbolic execution in K in general and in Kontrol in particular is performed on a symbolic state consisting of: a **configuration**, which is a collection of cells; and a **path condition**, which is a set of Boolean constraints on the symbolic variables present in the configuration.
+
+The purpose of simplifications, as their name suggests, is to simplify the job of the symbolic execution engine. They describe ways of getting the configuration and/or the path condition into a preferred format or state mathematical theorems that are either too complicated for the SMT solver to reason about both automatically and tractably or too simple for us to even want to call the SMT solver. Simplifications are K rewrite rules that have the `[simplification]` attribute, and their general form is:
+
+```k
+rule LHS => RHS requires C1 andBool ... andBool CN [simplification]
+```
+
+meaning that all occurrences of the term `LHS` in the current symbolic state should be replaced with the term `RHS` when the Boolean constraints `C1`, ..., `CN` are entailed by the current path condition `PC`, that is, when $\mathtt{PC} \implies \mathtt{C1} \wedge \ldots \wedge \mathtt{CN}$.
+
+Importantly, a simplification is **sound** if and only if
+$$
+\mathtt{C1} \wedge \ldots \wedge \mathtt{CN} \implies \mathtt{(LHS \Leftrightarrow RHS)}
+$$
```

Comment:
This seems not to be rendering nicely, at least on the plain Github markdown render (see image below). Is it possible that $$ is not supported? Or that maybe has to be done in the `$$\mathtt{C1} \wedge \ldots \wedge \mathtt{CN} \implies \mathtt{(LHS \Leftrightarrow RHS)}$$` style?
Or maybe gitbook will render this appropriately.
Image below:
![image](https://github.com/user-attachments/assets/d0fc19c8-834c-4775-8fc5-563d6200cdd2)

Reply:
I checked in a markdown preview plugin, and it renders fine. So I think it's a Github markdown limitation, I'll merge it as is and will then push a follow up PR if it doesn't render well there.